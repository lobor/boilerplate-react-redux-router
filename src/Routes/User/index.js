import React from 'react'
import { routeStore, navigate } from 'store/router'
import { Link, Row, Table, Box } from 'components';

export default class User extends React.Component {
	componentWillMount() {
		this.props.list();
	}

	openUser(id) {
		routeStore.dispatch(navigate('/user/'+id))
	}

	render () {
		return (
			<Row>
				<Box>
					<Table>
						<thead>
							<tr>
								<th>Id</th>
								<th>First Name</th>
								<th>Last Name</th>
							</tr>
						</thead>
						<tbody>
							{this.props.users.map((user, key) => {
								return (
									<tr key={key} onClick={() => { this.openUser(user.id) }}>
										<td>
											{user.id}
										</td>
										<td>
											{user.firstName}
										</td>
										<td>
											{user.lastName}
										</td>
									</tr>
								)
							})}
						</tbody>
					</Table>
				</Box>
			</Row>
		)
	}
}
