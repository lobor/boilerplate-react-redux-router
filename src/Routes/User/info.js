import React from 'react'
import { Input, Row, Box, Column } from 'components';

export default class UserInfo extends React.Component {
	componentWillMount() {
		this.props.get(parseInt(this.props.location.params.id, 10));
	}

	render () {
		var { lastName, firstName } = this.props.user
		return (
			<Row>
				<Column>
					<label htmlFor="firstName">First Name</label>
					<Input type="text" value={firstName} id="firstName" name="firstName"/>
				</Column>
				<Column>
					<label htmlFor="lastName">Last Name</label>
					<Input type="text" value={lastName} id="lastName" name="lastName"/>
				</Column>
			</Row>
		)
	}
}
