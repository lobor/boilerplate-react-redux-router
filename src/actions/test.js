export function get(title) {
	return function (dispatch) {
		new Promise((resolve) => {
			dispatch( {
		    type: 'test',
		    title: title || 'hello'
		  })
		})
	}
};
