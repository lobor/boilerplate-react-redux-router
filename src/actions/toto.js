export function get(title = 'hello') {
	return {
    type: 'test',
    payload: { title }
	}
};
