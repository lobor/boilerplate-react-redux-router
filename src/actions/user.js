export function get(user = {}) {
	return {
    type: 'test',
    payload: { user }
	}
};

export function list(users = []) {
	return {
    type: 'test',
    payload: { users }
	}
};
