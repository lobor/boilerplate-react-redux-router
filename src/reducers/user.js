var initialState = {
	users: [
		{
			id: 1,
			lastName: 'Dupond',
			firstName: 'Lionel'
		},
		{
			id: 2,
			lastName: 'Tralfalgar',
			firstName: 'Nami'
		}
	],
	user: {}
};

export default function user(state = initialState, { type, payload }) {
  switch (type) {
    case 'LIST_USER':
      return state;
		case 'GET_USER':
			var user;
			state.users.forEach(function (el) {
				if (el.id === payload.id){
					user = el
				}
			})
      return Object.assign(state, { user: user });
    default:
      return state
  }
}
