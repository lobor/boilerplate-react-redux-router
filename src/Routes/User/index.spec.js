import React from 'react'
import renderer from 'react-test-renderer'
import { list } from 'actions/user'
import User from './index'
import { shallow } from 'enzyme';

const users = [
	{
		id: 1,
		lastName: 'Dupond',
		firstName: 'Lionel'
	}
]

describe('Test route User', () => {
	const component = (<User list={list} users={users} />)
	test('render User', () => {
	  expect(component).toMatchSnapshot()
	})

	// test('click on user', () => {
	// 	console.log(component);
	// 	var toto  = shallow(component)
	// 	toto.find('tbody tr').simulate('click');
	// 	// component.openUser(1);
	//   // expect(tree).toMatchSnapshot()
	// })
})
