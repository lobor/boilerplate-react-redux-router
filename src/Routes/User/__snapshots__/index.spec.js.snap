// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test route User render User 1`] = `
<User
  list={[Function]}
  users={
    Array [
      Object {
        "firstName": "Lionel",
        "id": 1,
        "lastName": "Dupond",
      },
    ]
  }
/>
`;
